saidump/saidump.cpp

```diff
@@ -399,7 +417,170 @@ void dumpGraph(const TableDump& td)
     std::cout << "}" << std::endl;
 }
 
-int main(int argc, char ** argv)
+/*
+ preprocess the input json file to make sure it's a valid json file.
+*/
+static int preProcessFile(std::string file_name)
+{
+    std::ifstream input_file(file_name);
+
+    if (!input_file.is_open())
+    {
+        std::cerr << "Failed to open the input file " << file_name << std::endl;
+        SWSS_LOG_ERROR("Failed to open the input file %s.", file_name.c_str());
+        return SAI_STATUS_FAILURE;
+    }
+
+    input_file.seekg(0, std::ios::end);     // Move to the end of the file
+    int64_t file_size = input_file.tellg(); // Get the current position
+
+    if (file_size >= RDB_FILE_MAX_SIZE)
+    {
+        std::cerr << "Get " << file_name.c_str() << "'s size failure or its size " << file_size << "  >= "<< RDB_FILE_MAX_SIZE / 1024 / 1024<<" MB." << std::endl;
+        SWSS_LOG_ERROR("Get %s's size failure or its size %ld  >= %ld MB.", file_name.c_str(), file_size, RDB_FILE_MAX_SIZE / 1024 / 1024);
+        // Close the input file
+        input_file.close();
+        return SAI_STATUS_FAILURE;
+    }
+
+    input_file.seekg(0, std::ios::beg);     // Move to the begin of the file
+
+    // Read the content of the input file into a string
+    std::string content((std::istreambuf_iterator<char>(input_file)),
+                   std::istreambuf_iterator<char>());
+    input_file.close();
+
+    content = regex_replace(content, std::regex("\\},\\{\\r"), ",");
+
+    //erase the 1st and last char.
+    if (content.length() >= 2 && content[0] == '[' && content[content.length()-1] == ']')
+    {
+        content.erase(0, 1);
+        content.erase(content.length() - 1);
+    }
+
+    std::ofstream outputFile(file_name);
```

Comment:
you are reading and writting to the same file ? i thought you read the input database and then just process and output to stdout ?

Reply:
The system's current nlohmann::json library doesn't correctly parse the Json file's format of rdbtool's output. So I used the function preProcessFile to process it before using  nlohmann::json library. If the  nlohmann::json library was updated, preProcessFile would be removed.

Reply:
nlohmann::json is actually being updated soon